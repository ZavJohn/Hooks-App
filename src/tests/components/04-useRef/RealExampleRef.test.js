import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';


describe('Pruebas del <RealExampleRef />', () => {
    
    const wrapper = shallow( <RealExampleRef /> );

    test('Debe mostrarse correctamente', () => { 
           
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(false);
     });

     test('Debe mostrar la informaciÃ³n', () => { 

        wrapper.find('button').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true);
        
     });



});