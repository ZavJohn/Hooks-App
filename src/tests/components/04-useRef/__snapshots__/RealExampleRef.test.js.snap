// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del <RealExampleRef /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Ejemplos del uso real de Ref
  </h2>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show / Hide
  </button>
</Fragment>
`;
